mne/viz/topo.py

```diff
@@ -713,22 +706,28 @@ def _plot_evoked_topo(evoked, layout=None, layout_scale=0.945, color=None,
 
     if not merge_channels:
         # XXX. at the moment we are committed to 1- / 2-sensor-types layouts
-        chs_in_layout = set(layout.names) & set(ch_names)
-        types_used = {channel_type(info, ch_names.index(ch))
-                      for ch in chs_in_layout}
+        chs_in_layout = [ch_name for ch_name in ch_names
+                         if ch_name in layout.names]
+        # Using List comprehension to order elements and prevent randomness
+        types_used = [channel_type(info, ch_names.index(ch))
+                      for ch in chs_in_layout]
+        # Using dict conversion to remove duplicates
+        types_used = list(dict.fromkeys(types_used))
         # remove possible reference meg channels
-        types_used = set.difference(types_used, set('ref_meg'))
+        types_used = [types_used for types_used in types_used
+                      if types_used != 'ref_meg']
         # one check for all vendors
-        meg_types = {'mag', 'grad'}
-        is_meg = len(set.intersection(types_used, meg_types)) > 0
-        nirs_types = {'hbo', 'hbr', 'fnirs_cw_amplitude', 'fnirs_od'}
-        is_nirs = len(set.intersection(types_used, nirs_types)) > 0
+        is_meg = len([x for x in types_used if x in ['mag', 'grad']]) > 0
+        is_nirs = len([x for x in types_used if x in
+                      ['hbo', 'hbr', 'fnirs_cw_amplitude', 'fnirs_od']]) > 0
         if is_meg:
-            types_used = list(types_used)[::-1]  # -> restore kwarg order
+            # Fix issue where ylimits get swapped
+            types_used = list(types_used)[::-1]
```

Comment:
this is still needed? I'd remove it if not. Could be confusing in the future

Reply:
Yeah, That line existed before this PR and is still required by default